File: Java/QuizGame/src/utils/QuestionReader.java
package utils;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class QuestionReader {
private String path;
	
	public QuestionReader(String path) {
		this.path = path;
	}
	public ArrayList<String> getDomanda(int nDomanda){
		ArrayList<String> res = new ArrayList<>();
		File file = new File(path+"\\"+nDomanda+".txt");
	    Scanner sc;
		try {
			sc = new Scanner(file);
			for(int i=0;i<5;i++) {
				res.add(sc.nextLine());
			}
			sc.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
		
		return res;
	}
}

File: C++/QuizGame/src/utils/QuestionReader.cpp
#include "QuestionReader.h"
#include <fstream>
#include <vector>

QuestionReader::QuestionReader(std::string path) {
	this->path = path;
}

std::vector<std::string> QuestionReader::getDomanda(int nDomanda){
	std::fstream file;
	std::string temp;
	std::vector<std::string> res;

	file.open((path+"/"+std::to_string(nDomanda)+".txt"), std::ios::in);

	for(int i=0;i<5;i++){
		if(file.is_open()){
			std::getline(file,temp);
		}
		res.push_back(temp);
	}

	file.close();
	return res;
}

QuestionReader::~QuestionReader() {
}

